flake.nix

```
@@ -17,38 +17,81 @@
 
       ci-agent = {
         url = "github:hercules-ci/hercules-ci-agent";
-        inputs = { nix-darwin.follows = "darwin"; flake-compat.follows = "flake-compat"; nixos-20_09.follows = "nixos"; nixos-unstable.follows = "override"; };
+        inputs = { nix-darwin.follows = "darwin"; flake-compat.follows = "flake-compat"; nixos-20_09.follows = "nixos"; nixos-unstable.follows = "latest"; };
       };
       darwin.url = "github:LnL7/nix-darwin";
-      darwin.inputs.nixpkgs.follows = "override";
+      darwin.inputs.nixpkgs.follows = "latest";
       flake-compat.url = "github:BBBSnowball/flake-compat/pr-1";
       flake-compat.flake = false;
       home.url = "github:nix-community/home-manager";
       home.inputs.nixpkgs.follows = "nixos";
       naersk.url = "github:nmattia/naersk";
-      naersk.inputs.nixpkgs.follows = "override";
+      naersk.inputs.nixpkgs.follows = "latest";
       nixos-hardware.url = "github:nixos/nixos-hardware";
 
       pkgs.url = "path:./pkgs";
       pkgs.inputs.nixpkgs.follows = "nixos";
     };
 
-  outputs = inputs@{ self, devos, nixos, nur, ... }:
+  outputs = inputs@{ self, pkgs, devos, nixos, ci-agent, home, nixos-hardware, nur, ... }:
+
     devos.lib.mkFlake {
       inherit self inputs nixos;
-      hosts = ./hosts;
-      packages = import ./pkgs;
-      suites = import ./profiles/suites.nix;
-      extern = import ./extern;
-      overrides = import ./extern/overrides.nix;
-      overlays = ./overlays;
-      profiles = ./profiles;
-      userProfiles = ./users/profiles;
-      modules = import ./modules/module-list.nix;
-      userModules = import ./users/modules/module-list.nix;
-    } // {
-    defaultTemplate = self.templates.flk;
-    templates.flk.path = ./.;
-    templates.flk.description = "flk template";
-  };
+
+      channelsConfig = { allowUnfree = true; };
+
+      channels = {
+        nixos = {
+          overlays = [
+            ./overlays
+            pkgs.overlay
+            ./overrides.nix # from "latest" channel
+          ];
+          externalOverlays = [
+            nur.overlay
+          ];
+        };
+        latest = { };
+      };
+
+      os = {
+        hostDefaults = {
+          system = "x86_64-linux";
+          channelName = "nixos";
+          modules = ./modules;
```

```suggestion
          modules = ./modules/module-list.nix;
```
this needs to be a list of modules, modules aren't auto imported. And for the same reason, you can't delete module-list.nix.

---

Ah true, this still represents my state of mind some two days ago :wink: